lti_consumer/plugin/views.py

```diff
@@ -416,3 +425,196 @@ def scores(self, request, *args, **kwargs):  # pylint: disable=unused-argument
             status=status.HTTP_201_CREATED,
             headers=headers
         )
+
+
+class LtiNrpsContextMembershipViewSet(viewsets.ReadOnlyModelViewSet):
+    """
+    LTI NRPS Context Membership Service endpoint.
+    """
+
+    # Custom permission classes for LTI APIs
+    authentication_classes = [Lti1p3ApiAuthentication]
+    permission_classes = [LtiNrpsContextMembershipsPermissions]
+
+    # Renderer classes to accept LTI NRPS content types
+    renderer_classes = [
+        MembershipResultRenderer,
+    ]
+
+    # Custom pagination class to support pagination via Link Header
+    pagination_class = LinkHeaderPagination
+
+    @staticmethod
+    def parse_role_filter(nrps_filter_role=None):
+        """
+        Parse role filter from query param and map it to edx role.
+        """
+        if nrps_filter_role:
+            for edx_role, context_roles in LTI_1P3_CONTEXT_ROLE_MAP.items():
+                if nrps_filter_role in context_roles:
+                    return edx_role
+
+        # if no NRPS filter role provided or there was no match in LTI_1P3_CONTEXT_ROLE_MAP
+        return None
+
+    @staticmethod
+    def build_filter_kwargs(filter_by=None):
+        """
+        Helper method to build filter arguments for get_course_membership python API
+        Returns:
+            access_roles (list): A list of roles if there any, otherwise None
+            include_students (bool)
+        """
+        access_roles = None
+        include_students = True
+
```

Comment:
```suggestion
        role_filter = self.map_lti_to_lms_role(lti_role)

        access_roles = None
        include_students = True
```

Reply:
@giovannicimolin, since this is a static method we don't have `self` here. Do you want those as regular methods instead?

Reply:
@shimulch Can you make only `get_filter_params` a regular method? You can leave the other one as static.